SimTKcommon/Scalar/include/SimTKcommon/internal/NTraits.h

```diff
@@ -799,9 +799,14 @@ template <class R> class NTraits< conjugate<R> > {
     static TNormalize normalize(const T& t) {return TNormalize(t/abs(t));}
 
     // 1/conj(z) = conj(1/z), for complex z.
-    static TInvert invert(const T& t)    
-    {   const typename NTraits<THerm>::TInvert cmplx(NTraits<THerm>::invert(t.conj()));
-        return reinterpret_cast<const TInvert&>(cmplx); } // recast complex to conjugate it
+    // Here reinterpret_cast is source of [-Wstrict-aliasing] warning.
+    // New code is more explicit and should be faster(???).
+    static TInvert invert(const T& t){
+      return static_cast<TInvert>(TReal(1)/t.conj());
```

Comment:
The point of the above method is to conjugate the complex quotient (1/t.conj) merely by reinterpreting it as being of type `conjugate`, rather than performing the conjugation operation, which costs a floating point operation. That's why it does a reinterpret_cast, which is correct. Then the copy constructor is invoked to return the result with no computation.

In the new version, I believe the complex quotient is converted to a conjugate type *of the same value*, which means it is the conjugate of the result rather than the actual result. (That is, this returns with the imaginary part negated from the correct answer.)

So I think you got rid of the warning by making the code wrong! Unfortunately it appears there wasn't a regression test for this -- there should be and you definitely would have to add some tests if you want to mess with this very touchy and performance-sensitive code. It would be great to see some better tests for these operations.